database/database_test.go

```diff
@@ -234,3 +234,180 @@ func TestStatesWithStatus(t *testing.T) {
 		}
 	}
 }
+
+// setTests is a struct that is used in testing SetIneligibleDomainStates
+var setTests = []struct {
```

Comment:
The name `setTests` is too generic ("set" is a common db operation, while this is specific to IneligibleDomainStates). Either rename it to be more specific, or move it inside of TestSetIneligibleDomainStates.